domain/infosync/info_test.go

```diff
@@ -20,172 +20,127 @@ import (
-	resp, err := is.etcdCli.Get(ctx, key)
-	if err != nil {
-		return nil, err
-	}
-	if len(resp.Kvs) == 0 {
-		return nil, errors.New("not-exists")
-	}
-	if len(resp.Kvs) != 1 {
-		return nil, errors.New("resp.Kvs error")
-	}
-	var ret topologyInfo
-	err = json.Unmarshal(resp.Kvs[0].Value, &ret)
-	if err != nil {
-		return nil, err
-	}
-	return &ret, nil
-}
-
-func (is *InfoSyncer) ttlKeyExists(ctx context.Context) (bool, error) {
-	key := fmt.Sprintf("%s/%s:%v/ttl", TopologyInformationPath, is.info.IP, is.info.Port)
-	resp, err := is.etcdCli.Get(ctx, key)
-	if err != nil {
-		return false, err
-	}
-	if len(resp.Kvs) >= 2 {
-		return false, errors.New("too many arguments in resp.Kvs")
+func TestMain(m *testing.M) {
+	testbridge.WorkaroundGoCheckFlags()
+	opts := []goleak.Option{
+		goleak.IgnoreTopFunction("go.etcd.io/etcd/pkg/logutil.(*MergeLogger).outputLoop"),
 	}
-	return len(resp.Kvs) == 1, nil
-}
-
-func TestT(t *testing.T) {
-	CustomVerboseFlag = true
-	TestingT(t)
-}
-
-var _ = Suite(&testSuite{})
-
-type testSuite struct {
+	goleak.VerifyTestMain(m, opts...)
 }
 
 func TestTopology(t *testing.T) {
 	if runtime.GOOS == "windows" {
 		t.Skip("integration.NewClusterV3 will create file contains a colon which is not allowed on Windows")
 	}
+
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	currentID := "test"
 
-	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1})
-	defer clus.Terminate(t)
+	cluster := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1})
+	defer cluster.Terminate(t)
 
-	cli := clus.RandClient()
+	client := cluster.RandClient()
 
-	err := failpoint.Enable("github.com/pingcap/tidb/domain/infosync/mockServerInfo", "return(true)")
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/domain/infosync/mockServerInfo", "return(true)"))
 	defer func() {
 		err := failpoint.Disable("github.com/pingcap/tidb/domain/infosync/mockServerInfo")
-		if err != nil {
-			t.Fatal(err)
-		}
+		require.NoError(t, err)
 	}()
 
-	info, err := GlobalInfoSyncerInit(ctx, currentID, func() uint64 { return 1 }, cli, false)
-	if err != nil {
-		t.Fatal(err)
-	}
+	info, err := GlobalInfoSyncerInit(ctx, currentID, func() uint64 { return 1 }, client, false)
+	require.NoError(t, err)
 
 	err = info.newTopologySessionAndStoreServerInfo(ctx, owner.NewSessionDefaultRetryCnt)
+	require.NoError(t, err)
 
-	if err != nil {
-		t.Fatal(err)
-	}
+	topology, err := info.getTopologyFromEtcd(ctx)
+	require.NoError(t, err)
+	require.Equal(t, int64(1282967700000), topology.StartTimestamp)
 
-	topo, err := info.getTopologyFromEtcd(ctx)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if topo.StartTimestamp != 1282967700000 {
-		t.Fatal("start_timestamp of topology info does not match")
-	}
-	if v, ok := topo.Labels["foo"]; !ok || v != "bar" {
-		t.Fatal("labels of topology info does not match")
```

Comment:
Maybe keep these comments.

Reply:
Do you mean exactly this comment or all comments in this file? I think assertion itself provides a few information to understand what fails.

Reply:
Makes sense and acceptable, resolved.